components/DataTable/src/DataTable/DataTable.cs

```diff
@@ -93,7 +93,11 @@ protected override Size MeasureOverride(Size availableSize)
                 // then invalidate the child arranges [don't re-measure and cause loop]...)
 
                 // For now, we'll just use the header content as a guideline to see if things work.
-                column.Measure(new Size(availableSize.Width - fixedWidth - autoSized, availableSize.Height));
+
+                // Avoid negative values when columns don't fit `availableSize`. Otherwise the `Size` constructor will throw.
+                double width = availableSize.Width - fixedWidth - autoSized;
+                width = Math.Max(width, 0);
+                column.Measure(new Size(width, availableSize.Height));
```

Comment:
```suggestion
                column.Measure(new Size(Math.Max(availableSize.Width - fixedWidth - autoSized, 0), availableSize.Height));
```
It'll probably get optimized, but might as well just do it inline vs. splitting it all out, this is a common enough practice for clamping.